src/main/ipc/docker.ts

```diff
@@ -0,0 +1,356 @@
+  const dockerClient = new Dockerode();
+  log(`Pull ${image}`);
+  const pullStream = await dockerClient.pull(image);
+  // logger.info(`----------- pullStream ${JSON.stringify(pullStream)}`);
+
+  function onPullProgress(_event: any) {
+    // ...
+    // logger.info(`onPullProgress: ${JSON.stringify(_event)}`);
+    log(`onPullProgress: ${JSON.stringify(_event)}`);
+  }
+
+  await new Promise((resolve) =>
+    dockerClient.modem.followProgress(pullStream, resolve, onPullProgress)
+  );
+
+  log(`finished Pulling ${image}`);
+
+  return dockerClient
+    .createContainer({
+      name: 'solana-test-validator',
+      Image: image as string,
+      AttachStdin: false,
+      AttachStdout: true,
+      AttachStderr: true,
+      Tty: true,
+      Entrypoint: 'tail',
+      Cmd: ['-f', '/etc/os-release'],
+      OpenStdin: false,
+      StdinOnce: false,
+      Labels: {
+        environment: 'blueWhale',
+      },
+      ExposedPorts: {
+        '8899/tcp': {},
+        '8900/tcp': {},
+      },
+      HostConfig: {
+        PortBindings: {
+          '8899/tcp': [
+            {
+              HostPort: '8899',
+            },
+          ],
+          '8900/tcp': [
+            {
+              HostPort: '8900',
+            },
+          ],
+        },
+      },
+    })
+    .then((container: Dockerode.Container) => {
+      logger.info(`container created ${container.id}`);
+      log('container created');
+      return 'OK';
+    })
+    .catch(logger.error);
+}
+
+async function execAmman() {
+  // TODO: this presupposes that this workbench session starts the validator
+  //       should change this so the `amman start` tee's to a file, and then use `docker exec tail -n 20` or something
+  copyDockerFile({
+    filename: '.ammanrc.js',
+    content: `module.exports = ${JSON.stringify(ammanrc)}`,
+    destination: '/test-ledger',
+  });
+
+  const dockerClient = new Dockerode();
+  const container = dockerClient.getContainer('solana-test-validator');
+  logger.error(`request start amman exec: solana-test-validator`);
+  log(`request start amman exec: solana-test-validator`);
+
+  const logStream = new newStream.PassThrough();
+  logStream.on('data', (chunk) => {
+    log(chunk.toString('utf8'));
+  });
+
+  const p = container
+    .exec({
+      Cmd: ['/bin/bash', '-c', 'source /root/.bashrc && amman start'],
+      AttachStdin: true,
+      AttachStdout: true,
+    })
+    .then((exec: Dockerode.Exec) => {
+      log('start amman exec created');
+
+      return exec.start({
+        // hijack: true,
+        // stdin: true,
+        // Detach: false,
+        // Tty: true,
+      });
+    });
+
+  const stream = await p;
+  container.modem.demuxStream(stream, logStream, logStream);
+  log('Exiting execAmman');
+  return p;
+}
+
+// Need to import the file and call a function (from the main process) to get the IPC promise to exist.
+export function initDockerPromises() {
+  promiseIpc.on(
+    'DOCKER-GetImageTags',
```

Comment:
can merge this in as-is, but in future, can we stick to existing `skewer-case` convention please?